components/kfp/Run_component/component.py

```diff
@@ -0,0 +1,47 @@
+from typing import NamedTuple
+
+
+def run_component_or_pipeline(
+    component_url: 'Url',
+    arguments: dict,
+    endpoint: str = None,
+    wait_timeout_seconds: float = None,
+) -> NamedTuple('Outputs', [
+    ('run_id', str),
+    ('run_object', 'JsonObject'),  # kfp.ApiRunDetails
+]):
+    import json
+    import os
+    import kfp
+    from kfp_server_api import ApiClient
+    print('Loading component...')
+    op = kfp.components.load_component_from_url(component_url)
```

Comment:
This looks awesome! I'd like to use this component for https://github.com/kubeflow/pipelines/pull/5337.

However, how does this support running a pipeline? graph component?

After thinking about my use-case, I think a few options would enable my scenario:
1. I want to load a component/pipeline from local path too (passed from input artifacts).
2. I want to get run status also as an explicit output parameter.
3. the script can get very complex with many options, consider always adding unit tests next to a component?

What do you think? Note, I might just go ahead and implement a fork of this component to satisfy immediate needs quickly, but I think it'll be worth it standardizing what we do for our own tests as a reusable component that everyone can use to set up their pipeline CI/CD.

Reply:
Thank you =)

>However, how does this support running a pipeline? graph component?

Graph component is an option.
But I also think we should add more `kfp` components alongside. For example, a component to submit a pipeline from python code. `Run_pipeline`.

>I want to load a component/pipeline from local path too (passed from input artifacts).

Good idea. I should probably split this component into `from_file` and `from_url`.

>I want to get run status also as an explicit output parameter.

Good idea. I'll add that.

>the script can get very complex with many options, consider always adding unit tests next to a component?

What would be an action item here? I usually add sample pipelines for every component or a group of components.

Reply:
> For example, a component to submit a pipeline from python code. Run_pipeline.

I think there are a lot of user interest for this too.

> Good idea. I should probably split this component into from_file and from_url.

I'd prefer sharing most of the code, but allow two optional inputs, one for from file and one for from url. One single component that works for many scenarios sound easier to understand -- think about unix philosophy of building tools that are small, focused and does a lot of things flexibly. It's harder to manage a larger set of tools.

Reply:
my example: https://github.com/kubeflow/pipelines/pull/5337/files#diff-9cf6e1982d0f982025dd3d53d87a82aa46b313b78511769cf66629aa22e36fbdR4-R15

do you think that's easier to understand than two separate components?

Reply:
> What would be an action item here? I usually add sample pipelines for every component or a group of components.

AI: add presubmit test that runs unit test for all component unit tests. I think it'll be great if we build components following all the engineering best practices -- including CI.

Reply:
One thing I don't have a good idea right now is that, I need to set component image dynamically and use a pipeline from a local path: https://github.com/kubeflow/pipelines/blob/fc769b82a752b5680a524bd4e488e634ef04aa23/v2/test/sample_test.py#L50

How do I do that with this component? or what pattern should I switch to?

Reply:
>AI: add presubmit test that runs unit test for all component unit tests. I think it'll be great if we build components following all the engineering best practices -- including CI.

Sure. Let's start doing that once your new sample test infra gets in.

Reply:
>I'd prefer sharing most of the code, but allow two optional inputs, one for from file and one for from url.

This does not always work (e.g. for TFX components), but I guess for this component it's OK.

P.S. It's a bit confusing when optional inputs are not really optional (one must be provided).